src/edu/stanford/dlss/wowza/SulWowza.java

```diff
@@ -83,22 +86,40 @@ public void onHTTPCupertinoStreamingSessionCreate(HTTPStreamerSessionCupertino h
         }
     }
 
-    /** Invoked when an Adobe HDS streaming session is created (known as San Jose Streaming in Wowza parlance).
-     *  This is an HTTP-based (port 80) streaming protocol for Flash. We currently use this for streaming MP3
-     *  files, since there are no current HLS players that support MP3 across all browsers. */
-    public void onHTTPSanJoseStreamingSessionCreate(HTTPStreamerSessionSanJose httpSession)
-    {
+    /**
+     * Invoked when a media file starts playing. Defined in ModuleCore. This seems to be the only place
+     * we can reliably intercept a Flash connection and get the name of the stream.
+     */
+    public void play(IClient client, RequestFunction function, AMFDataList params)
+	{
+        String streamName = params.getString(PARAM1);
+
+        //get the real stream name if this is an alias.
+        streamName = ((ApplicationInstance)client.getAppInstance()).internalResolvePlayAlias(streamName, client);
+
         if (invalidConfiguration)
         {
-            getLogger().error(this.getClass().getSimpleName() + " onHTTPSanJoseinoStreamingSessionCreate: rejecting session due to invalid stacksURL property " + httpSession.getStreamName());
-            httpSession.rejectSession();
+            getLogger().error(this.getClass().getSimpleName() + " play: " + stacksUrlErrorMsg + "; streamName:  " + streamName);
+            client.shutdownClient();
         }
         else
         {
-          getLogger().info(this.getClass().getSimpleName() + " onHTTPSanJoseStreamingSessionCreate: " + httpSession.getStreamName());
-          authorizeSession(httpSession);
+            String queryString = client.getQueryStr();
+            if (queryString == null || queryString.length() == 0)
+                queryString = streamName;
```

Comment:
a comment explaining this `if` statement and when it's expected to be used might be nice... normally you'd expect `getQueryStr()` to do the right thing, but for whatever reason, it's returning `null` and you have to parse the stream name, right?  but IIRC from prior conversations, `getQueryStr` was never returning the correct result?